Source/WebKit/NetworkProcess/curl/NetworkDataTaskCurl.cpp

```diff
@@ -222,6 +226,8 @@ void NetworkDataTaskCurl::curlDidComplete(CurlRequest&, NetworkLoadMetrics&& net
         return;
     }
 
+    updateNetworkLoadMetrics(networkLoadMetrics);
```

Comment:
Cocoa port is calling checkTAO in willPerformHTTPRedirection and didReceiveResponse.
On the other hand, NetworkDataTaskCurl::updateNetworkLoadMetrics is called in curlDidComplete and curlDidReceiveResponse.
Is this right?

Reply:
This is right. Cocoa port store NetworkLoadMetrics in a member variable of the NetworkDataTaskCocoa class.

https://github.com/WebKit/WebKit/blob/5ab8b234cd6549d90d5d7853b340fbcb3646af97/Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.h#L115

On the other hand, Curl port does not store it and a new object is spawned every time when called CurlHandle::getNetworkLoadMetrics.

https://github.com/WebKit/WebKit/blob/5ab8b234cd6549d90d5d7853b340fbcb3646af97/Source/WebCore/platform/network/curl/CurlContext.cpp#L820

https://github.com/WebKit/WebKit/blob/5ab8b234cd6549d90d5d7853b340fbcb3646af97/Source/WebCore/platform/network/curl/CurlRequest.cpp#L387

https://github.com/WebKit/WebKit/blob/5ab8b234cd6549d90d5d7853b340fbcb3646af97/Source/WebCore/platform/network/curl/CurlRequest.cpp#L482

The generated objects are then notified to curlDidReceiveResponse and curlDidComplete. Therefore, NetworkDataTaskCurl::updateNetworkLoadMetrics must be called both curlDidReceiveResponse and curlDidComplete.